tactics/hints.ml

```diff
@@ -442,17 +448,22 @@ let rec path_derivate hp hint =
 let rec normalize_path h =
   match h with
   | PathStar PathEpsilon -> PathEpsilon
-  | PathSeq (PathEmpty, _) | PathSeq (_, PathEmpty) -> PathEmpty
-  | PathSeq (PathEpsilon, p) | PathSeq (p, PathEpsilon) -> normalize_path p
-  | PathOr (PathEmpty, p) | PathOr (p, PathEmpty) -> normalize_path p
   | PathOr (p, q) ->
-    let p', q' = normalize_path p, normalize_path q in
-      if hints_path_eq p p' && hints_path_eq q q' then h
-      else normalize_path (PathOr (p', q'))
+    (match normalize_path p with
+     | PathEmpty -> normalize_path q
+     | p' ->
+     match normalize_path q with
+     | PathEmpty -> p'
+     | q' -> if hints_path_eq p' q' then p' else PathOr (p', q'))
   | PathSeq (p, q) ->
-    let p', q' = normalize_path p, normalize_path q in
-      if hints_path_eq p p' && hints_path_eq q q' then h
-      else normalize_path (PathSeq (p', q'))
+    (match normalize_path p with
+     | PathEmpty -> PathEmpty
+     | PathEpsilon -> normalize_path q
+     | p' ->
+     match normalize_path q with
+     | PathEmpty -> p'
```

Comment:
This should read `PathEmpty`, not `p'`. Indeed, ` p · 0 ≡ 0`  in regexp semantics.

Reply:
Oh, right! Thanks a lot!